### ISSUE TITLE
Container ID outputted at `start` time confusing on the first pull

### ISSUE THREAD

**Issue by alexellis** [weaveworks/ignite#117]
I may be doing something wrong?

```
alex@nuc7:~$ sudo kvm-ok
INFO: /dev/kvm exists
KVM acceleration can be used
alex@nuc7:~$ sudo ignite run weaveworks/ignite-ubuntu --cpus 2 --memory 1024 --ssh --name my-vm
can't find kernel: no ID/name matches for "weaveworks/ignite-ubuntu"
alex@nuc7:~$ 

alex@nuc7:~$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.2 LTS (Bionic Beaver)"
```

**Comment by alexellis** [comment]
This is after running the help command:

```
Example usage:

	$ ignite run weaveworks/ignite-ubuntu \
		--cpus 2 \
		--memory 1024 \
		--ssh \
		--name my-vm
	$ ignite images
	$ ignite kernels
	$ ignite ps
	$ ignite logs my-vm
	$ ignite ssh my-vm
```

**Comment by twelho** [comment]
Could I see the output of `ignite version`?

**Comment by alexellis** [comment]
```
Ignite version: version.Info{Major:"0", Minor:"3", GitVersion:"v0.3.0", GitCommit:"9db63f66c8a38c83212d618f6d0d6995b79e07bf", GitTreeState:"clean", BuildDate:"2019-06-18T13:30:59Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Firecracker version: v0.16.0
```

Perhaps the instructions are missing a step to "pull a kernel"?

**Comment by alexellis** [comment]
```
alex@nuc7:~$ sudo ignite image ls
IMAGE ID		CREATED	SIZE		NAME
7f9feb5d10321f01	41m ago	320.5 MB	weaveworks/ignite-ubuntu:latest
alex@nuc7:~$ sudo ignite kernel ls
KERNEL ID	CREATED	SIZE	NAME
alex@nuc7:~$ 
```

**Comment by twelho** [comment]
In Ignite v0.4.0 the kernel is separated into it's own container unlike in previous versions, which have the filesystem and kernel bundled together. The `weaveworks/ignite-ubuntu:latest` image is already updated for v0.4.0, which means it doesn't contain a kernel anymore like v0.3.0 expects. Two options:
- Using your current version, add the tag `v0.2.0` to the `run` command like so:
```
$ ignite run weaveworks/ignite-ubuntu:v0.2.0 \
   ...
```
- Use version `v0.4.0-rc.1` of Ignite, which accepts just `weaveworks/ignite-ubuntu` as the image. Note that the `memory` flag takes in a string in v0.4.0, so use `1GB` instead of `1024`.

**Comment by alexellis** [comment]
New error using the version just pushed by @luxas 

```
INFO[0000] Docker image "centos:7" not found locally, pulling... 

INFO[0011] Starting image import...                     
INFO[0012] Imported OCI image "centos:7" (192.4 MB) to base image with UID "307f4c7bfe719ec6" 
INFO[0012] Docker image "weaveworks/ignite-kernel:4.19.47" not found locally, pulling... 
INFO[0017] Imported OCI image "weaveworks/ignite-kernel:4.19.47" (49.0 MB) to kernel image with UID "997e693af332bfd6" 
INFO[0018] Created VM with ID "80d1892909c0f961" and name "my-vm" 
INFO[0024] Started Firecracker VM "80d1892909c0f961" in a container with ID "Unable to find image 'weaveworks/ignite:v0.4.0' locally\nv0.4.0: Pulling from weaveworks/ignite\n10c7c21dc3bb: Pulling fs layer\n10c7c21dc3bb: Verifying Checksum\n10c7c21dc3bb: Download complete\n10c7c21dc3bb: Pull complete\nDigest: sha256:c345f65640f88c2e9e5c9e4c824792b0802758119ffd4b648a6c2555d469015f\nStatus: Downloaded newer image for weaveworks/ignite:v0.4.0\n9833bf617fc5e41718da5ef13521de6701f6f49dc731cc0386569cff790d950d"
```

**Comment by alexellis** [comment]
Very odd error! ^

2nd time:

```
alex@nuc7:~/go/src/github.com/openfaas/faas-netes$ sudo ignite run centos:7 --cpus 2 --memory 2GB --ssh --name my-vm-2
INFO[0001] Created VM with ID "4c58c7d4dcc8e298" and name "my-vm-2" 
INFO[0003] Started Firecracker VM "4c58c7d4dcc8e298" in a container with ID "5b4612de7394c8498b557a308291d338f285e817882b79430dd9162353762e47" 
```

ssh is not working however?

```
sudo ignite ssh my-vm-2
ssh: connect to host 172.17.0.3 port 22: Connection refused
SSH into VM "4c58c7d4dcc8e298" failed: external command "ssh root@172.17.0.3 -o StrictHostKeyChecking=no -i /var/lib/firecracker/vm/4c58c7d4dcc8e298/id_4c58c7d4dcc8e298" exited with an error: exit status 255
```

**Comment by luxas** [comment]
> INFO[0024] Started Firecracker VM "80d1892909c0f961" in a container with ID "Unable to find image 'weaveworks/ignite:v0.4.0' locally\nv0.4.0: Pulling from weaveworks/ignite\n10c7c21dc3bb: Pulling fs layer\n10c7c21dc3bb: Verifying Checksum\n10c7c21dc3bb: Download complete\n10c7c21dc3bb: Pull complete\nDigest: sha256:c345f65640f88c2e9e5c9e4c824792b0802758119ffd4b648a6c2555d469015f\nStatus: Downloaded newer image for weaveworks/ignite:v0.4.0\n9833bf617fc5e41718da5ef13521de6701f6f49dc731cc0386569cff790d950d"

Ugh, I thought I had fixed this... @twelho can you take a look?
Context is https://github.com/weaveworks/ignite/blob/master/pkg/operations/start.go#L70-L82

TL;DR; it's not an error, it just outputs stuff it shouldn't

@alexellis it's expected that `centos:7` does not support ssh, as the package is not installed by default in that docker image. Use `weaveworks/ignite-centos` instead and it works

**Comment by twelho** [comment]
Fixed with c5922ea, eventually we want to move to using `pkg/runtime` only and do any pulling required automatically, separating the output from the actual command output.
